boot/freeldr/freeldr/arch/uefi/uefildr.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * PROJECT:     Freeldr UEFI Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     UEFI Entry point and helpers
+ * COPYRIGHT:   Copyright 2022 Justin Miller <justinmiller100@gmail.com>
+ */
+
+#include <uefildr.h>
+
+#include <debug.h>
+
+/* GLOBALS ********************************************************************/
+
+EFI_HANDLE GlobalImageHandle;
+EFI_SYSTEM_TABLE *GlobalSystemTable;
+
+/* FUNCTIONS ******************************************************************/
+
+EFI_STATUS
+EfiEntry(
+    _In_ EFI_HANDLE ImageHandle,
+    _In_ EFI_SYSTEM_TABLE *SystemTable)
+{
+    SystemTable->ConOut->OutputString(SystemTable->ConOut, L"UEFI EntryPoint: Starting freeldr from UEFI");
+    GlobalImageHandle = ImageHandle;
+    GlobalSystemTable = SystemTable;
+
+    BootMain(NULL);
+
+    //UNREACHABLE
```

Comment:
We don't have the UNREACHABLE macro available here? If we do, then:
```suggestion
    UNREACHABLE;
```